Tests/SwiftParserTest/Assertions.swift

```diff
@@ -560,23 +564,42 @@ extension ParserTestCase {
     }
   }
 
-  /// Removes any test markers from `markedSource` (1) and parses the result
-  /// using `parse`. By default it only checks if the parsed syntax tree is
-  /// printable back to the origin source, ie. it round trips.
+  /// Verifies that parsing of `markedSource` produces expected results using a
+  /// combination of various testing techniques:
+  ///
+  /// 1. Asserts that parsing of `markedSource` round-trips, ie. that the printed
+  ///    parsed tree is the same as the input.
+  /// 2. Checks that parsing produces the expected list of diagnostics. If no
+  ///    diagnostics are passed, asserts that the input parses without any errors.
+  /// 3. Checks that applying all Fix-Its of the source code results in the
+  ///    expected fixed source, effectively testing the Fix-Its.
+  /// 4. If a substructure is passed, asserts that the parsed tree contains a
+  ///    subtree of that structure.
+  /// 5. Mutates the test input by flipping each token's presence (ie. for every
+  ///    token, remove it from the input if it is present in the parsed tree and
+  ///    synthesize a missing tokens into the input) and verifies that this
```

Comment:
```suggestion
  ///    synthesize any missing tokens into the input) and verifies that this
```

Reply:
This is intentionally `a` because only a single to ken will be synthesized not all of them.

Reply:
How about:
```
  /// 5. Mutates the test input by flipping each token's presence (ie. for every
  ///    token, either remove it from the input if it is present in the parsed tree or
  ///    synthesize it if it was missing) and verifies that this
```

Reply:
Sounds good